### ISSUE TITLE
Rebinning tool and resolution flag

### ISSUE THREAD

**Issue by NiallWhiteford** [SIMPLE-AstroDB/SIMPLE-db#274]
#### Low priority ####

1. Add tool to be able to bin data down to different resolution. 
2. We can add a "avg res" or "res range" flag for data being ingested. If this flag is included, long term, it means data can be grouped and queried in resolutions (low, mid, high) tiers.

**Comment by kelle** [comment]
I think you mean "field" or "column", right?  

Are you thinking of a quantitative R number?  Or a field with three options: low, medium, high?   If this were to be implemented, I would vote for an average R because low/med/high is very qualitative/relative and means something different to everyone.

**Comment by kelle** [comment]
I am a pretty hard no on the rebinning. We'd have to pay for that CPU power and folks can just download and rebin on their own. The goal of the database is to serve the data, not to be an analysis tool.

**Comment by kelle** [comment]
This has come up again because `species` needs it (#465).  

@tomasstolker said:

> The resolving power of a given instrument/mode is required for fitting model spectra. That information is currently not available in SIMPLE I think? Perhaps that is something to consider to include in Instruments.json. 

One challenge is that the SpeX prism data, for instance, has a different resolution as a function of wavelength. How accurate does it need to be?  Would average R be ok?

**Comment by tomasstolker** [comment]
I think that using an average resolution is fine, although I am used to working with low S/N spectra of directly imaged planets, so at high S/N such effects might become a bit more important. Perhaps for spectra that cross multiple bands (e.g. JHK), then it might be better to provide the average resolution per band and also the wavelength range for which the resolution is valid.

**Comment by kelle** [comment]
Yeah, this gets messy real fast. I think in order to do a proper analysis that involves spectral resolution, the investigator needs to calculate it themselves using the actual spectrum. The goal of SIMPLE is to facilitate *finding* the data, not the actual analysis. 

I think the goal of adding resolution to the Spectrum table would need to be to empower searches, not to facilitate analysis. E,g, find all the R>1000 spectra of L dwarfs.

**Comment by tomasstolker** [comment]
Okay, for searching purpose, the globally-averaged resolution should be sufficient indeed, which might often also be fine for fitting model spectra. Please correct me if I am wrong, but I don't think that the actual resolution (e.g. per band) can be easily determined from the data though. Perhaps by assuming Nyquist sampling, but this may not be true for all spectographs (e.g. SpeX spectra are oversampled if I remember correctly).

**Comment by kelle** [comment]
I think that just strengthens the point that it's too much to ask us (SIMPLE) to figure it out. Clearly beyond the scope.  But we can make `average_resolving_power` an optional field.

**Comment by tomasstolker** [comment]
Instead of the Spectrum table, perhaps this is easier to implement actually in the Instruments table? Since the resolving power will be the same for all spectra using a certain instrument and mode.
